lms/djangoapps/mobile_api/groups/views.py

```diff
@@ -0,0 +1,141 @@
+
+        {"success" : "true"}
+
+    """
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.GroupSerializer
+
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.DATA, files=request.FILES)
+        if serializer.is_valid():
+            post_args = {}
+            for key in request.POST.keys(): 
+                post_args[key] = request.POST[key]
+            try:
+                graph = facebook.GraphAPI(facebook.get_app_access_token(_FACEBOOK_APP_ID, _FACEBOOK_APP_SECRET))
+                app_groups_response = graph.request(_FACEBOOK_API_VERSION + _FACEBOOK_APP_ID + "/groups", post_args=post_args)
+            except Exception, e:
+                return Response({'error' : e.result['error']['message']}, status=status.HTTP_400_BAD_REQUEST)
+
+            return Response(app_groups_response)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+    def delete(self, request, *args, **kwargs):
+        if 'group_id' in kwargs:
+            post_args = {'method' : 'delete'}
+            graph = facebook.GraphAPI(facebook.get_app_access_token(_FACEBOOK_APP_ID, _FACEBOOK_APP_SECRET))
+            result = graph.request(_FACEBOOK_API_VERSION + _FACEBOOK_APP_ID + "/groups/" + kwargs['group_id'], post_args=post_args)
+            return Response(result)
+        else:
+            return Response({'error' : 'Missing group id'}, status=status.HTTP_400_BAD_REQUEST)
+
+
+
+class GroupsMembers(generics.CreateAPIView, mixins.DestroyModelMixin):
+    """
+    **Use Case**
+
+        An API to Invite and Remove members to a group
+
+    **Invite Example request**:
+
+        POST /api/mobile/v0.5/member/<group_id>/
+
+        Parameters: members : int,int,int... 
+
+
+    **Invite Response Values**
+
+        {"member_id" : success/error_message}       A response with each member_id and whether or not
+                                                    the member was added successfully. If the member was 
+                                                    not added successfully the Facebook error message is provided. 
+        
+    **Remove Example request**:
+
+        DELETE /api/mobile/v0.5/member/<group_id>/<member_id>
+
+    **Remove Response Values**
+
+        {"success" : "true"}                                    
+    """
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.GroupsMembersSerializer
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.DATA, files=request.FILES)
+        if serializer.is_valid():
+            graph = facebook.GraphAPI(facebook.get_app_access_token(_FACEBOOK_APP_ID, _FACEBOOK_APP_SECRET))
+            if 'group_id' in kwargs: 
```

Comment:
yes this is true, although group_id will always be in kwargs since it will not hit match the regex that maps to this view if a group_id was not provided. 
The same goes for every if x in kwargs 
I can see how this is confusing.